### PR TITLE
adding context from request example

### PR DESCRIPTION
**Pull request by lumost**
@cce 

adding an example for retrieving the context from an http request

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -187,6 +187,33 @@ func processRequest() {
...
+### Retrieving the context from an http request
+
+A common pattern when tracing in golang is to call `tv.HTTPHandler(handler)` then retrieve the trace
+context inside of the handler.  
+```go
+
+func myHandler( w http.ResponseWriter, r *http.Request ) {
+    // trace this request, overwriting w with wrapped ResponseWriter
+    t, w := tv.TraceFromHTTPRequestResponse("myHandler", w, r)
+    ctx := tv.NewContext(context.Background(), t)
```
Comment (MikeA-) [inline README.md]: @lumost I'm trying to follow along with the evolution of the examples we're giving, and as a non-go programmer things look much more complicated to me than they actually would to a go dev I'm sure.  I'm wondering why a call to NewContext is made, when the trace that was created earlier already has a context ( I think? ) that is stored in tvCtx and I think (?) can be accessed via https://github.com/appneta/go-appneta/blob/master/v1/tv/trace.go#L44

Or is there a subtlety here regarding context management for a trace?

Comment (lumost) [inline README.md]: @MikeA-

In this particular example myHandler only has access to the request, and the response.  In 
https://github.com/appneta/go-appneta/blob/master/v1/tv/http_instrumentation.go#L34

we attach the X-Trace header to the request object which allows us to construct a context inside of the handler.  This is necessary for http servers which directly use the standard libraries http.HandlerFunc type
https://golang.org/pkg/net/http/#HandlerFunc.ServeHTTP

alternative http frameworks which directly support context passing can use the same tracing context throughout the application - such as in our gin example

https://github.com/appneta/go-appneta/blob/master/examples/gin_app/gin_middleware.go#L23

Comment (cce) [inline README.md]: @MikeA- the context returned by tv.NewContext is a [context.Context](godoc.org/golang.org/x/net/context) interface type for use in your own Go program, and the NewContext helper is intended to help you the `tv.Trace` to a `context.Context` for passing around the user's own code — so the word "context" is being a bit overused here. The underlying TraceView metadata context in tvCtx is hidden from the user and is not being created twice — the call to `tv.NewContext` is basically binding the Trace (which contains a tvCtx and was already created, like you said) to a context.Context (which is a generic Golang package for use in request-handling).

**Comment by cce** [comment]
@lumost What do you think of the double-wrapping pattern — I actually just removed the double-wrapping using `tv.HTTPHandler` outside and then `tv.TraceFromHTTPRequestResponse` inside the handler from the alice and bob examples in https://github.com/appneta/go-appneta/commit/85dfa40ecadb5512cbcf0f7684b155a2603a882a#diff-85e8b4177a10f35a95aadde26d4683d6 because I thought it might be too confusing for the example to see the handler wrapped/traced twice — of course both ways work fine — should I put it back?

**Comment by cce** [comment]
After talking it over with @lumost wrapping the handler with `tv.HTTPHandler` even if you are going to call `tv.TraceFromHTTPRequestResponse` is harmless to do, and gets you Controller/Action names filled out with the method names of the handlers, so seems like a fine thing to recommend users do.

**Pull request merged**